clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1328,13 +1335,16 @@ RequestFuture<Void> sendOffsetCommitRequest(final Map<TopicPartition, OffsetAndM
             groupInstanceId = null;
         }
 
+        boolean canUseTopicIds = topicPartitionsWithoutTopicId == 0;
```

Comment:
Should we add tests to cover this new logic?

Reply:
Sure! Added the tests. Thanks.